### PR TITLE
doc: add comments for escape sequences in setup/restore terminal

### PR DESCRIPTION
**Pull request by Delapouite**
Hi

Most of the other ANSI escape sequences are already commented in the file, but these ones in the PR were missing.

### CODE REVIEW

File: src/terminal_ui.cc [hunk 1]
```diff
@@ -1250,9 +1250,13 @@ void TerminalUI::set_resize_pending()
 
 void TerminalUI::setup_terminal()
 {
+    // enable alternative screen buffer
     fputs("\033[?1049h", stdout);
+    // enable focus notify
     fputs("\033[?1004h", stdout);
+    // hide cursor
     fputs("\033[?25l", stdout);
+    // set special keypad mode
```
Comment (Screwtapello) [inline src/terminal_ui.cc:1259]: This is usually (i.e. in the original DEC terminal documentation) known as "application keypad" mode. If we're being explanatory, we might as well go all the way:
```suggestion
    // set application keypad mode, so the keypad keys send unique codes
```

File: src/terminal_ui.cc [hunk 2]
```diff
@@ -1263,6 +1267,7 @@ void TerminalUI::restore_terminal()
     fputs("\033[?1004l", stdout);
     fputs("\033[?25h", stdout);
     fputs("\033>", stdout);
+    // reset SGR
```
Comment (Screwtapello) [inline src/terminal_ui.cc:1270]: Again, we might as well explain all the way:
```suggestion
    // set the terminal output back to default colours and style
```

**Comment by lenormf** [comment]
I've actually been thinking we could make this more C++-y by storing the escape sequences as `constexpr` variables in the global scope (in a `struct` or whatever), which we would refer to by name, in the code.

The documentation added by this PR would be centralised where the sequences are declared.

**Comment by Delapouite** [comment]
@Screwtapello thanks for your expertise on this topic. I've updated the comments accordingly.
@lenormf Yes centralising this magic values in proper constants would be nice. But I don't feel comfortable doing these now. So a future PR can go this way.

**Comment by mawww** [comment]
cherry-picked.

**Pull request closed without merging**